### PR TITLE
feat(dependency): add specific agent protocol for dependency

### PR DESCRIPTION
**Pull request by Kafei59**
Linked to #57 

I set sync method config and parameters in a struct.
This way, I am able to load 2 types of providers, and choose during compute which one I want to use.
I have modified tests to check everything is OK.

Feel free to comment the way you wanted to do it initially

### CODE REVIEW

File: dependency/schema.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ type ProtoDepDependency struct {
 	Branch   string   `toml:"branch"`
 	Path     string   `toml:"path"`
 	Ignores  []string `toml:"ignores"`
+	Agent    string   `toml:"agent"`
```
Comment (holyhope) [inline dependency/schema.go:18]: Hello @Kafei59 :-)

```suggestion
	Protocol    string   `toml:"protocol"`
```

**Review by holyhope — APPROVED** [review]

**Comment by stormcat24** [comment]
@Kafei59 Thanks for your contribution.
I have confirmed your implementation. Very nice work!

@holyhope Thanks for helping the review!

**Comment by vigoo** [comment]
Hi! We would love to use this new feature, are there any plans for a release?

**Comment by stormcat24** [comment]
@vigoo This feature was released as [0.1.2](https://github.com/stormcat24/protodep/releases/tag/0.1.2)!

**Comment by vigoo** [comment]
Awesome, thank you!

**Pull request merged**